src/socket-to-conn.js

```diff
@@ -0,0 +1,65 @@
+'use strict'
+
+const abortable = require('abortable-iterator')
+const { CLOSE_TIMEOUT } = require('./constants')
+const toMultiaddr = require('libp2p-utils/src/ip-port-to-multiaddr')
+
+const log = require('debug')('libp2p:websockets:socket')
+
+// Convert a stream into a MultiaddrConnection
+// https://github.com/libp2p/interface-transport#multiaddrconnection
+module.exports = (socket, options = {}) => {
+  const maConn = {
+    async sink (source) {
+      if (options.signal) {
+        source = abortable(source, options.signal)
+      }
+
+      try {
+        await socket.sink(source)
+      } catch (err) {
+        if (err.type !== 'aborted') {
+          log(err)
```

Comment:
Should we make this `log.error(err)`? Makes it easier to just target errors with debugging on.